pysteps/blending/linear_blending.py

```diff
@@ -180,24 +184,70 @@ def forecast(
             # Set weights at times before start_blending and after end_blending
             if weight_nwp < 0.0:
                 weight_nwp = 0.0
+                if n_ens_members_max == 1:
+                    R_blended[i, :, :] = precip_nowcast[i, :, :]
+                else:
+                    R_blended[:, i, :, :] = precip_nowcast[:, i, :, :]
             elif weight_nwp > 1.0:
                 weight_nwp = 1.0
-
-            # Calculate weight_nowcast
-            weight_nowcast = 1.0 - weight_nwp
-
-            # Calculate output by combining precip_nwp and precip_nowcast,
-            # while distinguishing between ensemble and non-ensemble methods
-            if n_ens_members_max == 1:
-                R_blended[i, :, :] = (
-                    weight_nwp * precip_nwp[i, :, :]
-                    + weight_nowcast * precip_nowcast[i, :, :]
-                )
+                if n_ens_members_max == 1:
+                    R_blended[i, :, :] = precip_nwp[i, :, :]
+                else:
+                    R_blended[:, i, :, :] = precip_nwp[:, i, :, :]
             else:
-                R_blended[:, i, :, :] = (
-                    weight_nwp * precip_nwp[:, i, :, :]
-                    + weight_nowcast * precip_nowcast[:, i, :, :]
-                )
+                # Calculate weight_nowcast
+                weight_nowcast = 1.0 - weight_nwp
+
+                # Calculate output by combining precip_nwp and precip_nowcast,
+                # while distinguishing between ensemble and non-ensemble methods
```

Comment:
I feel this should not be necessary, but I don't have an obvious answer yet, will think about it!

Reply:
I think we can should be able to use the solution from https://stackoverflow.com/a/24399139/4222370 , something like
```
# 
if n_ens_members_max == 1:
   ref_dim = 0
else:
   ref_dim = 1

# compute the slice indices
def _get_slice(n_dims, ref_dim, ref_id):
  slc = [slice(None)] * n_dims
  slc[ref_dim] = ref_id
  return tuple(slc)

# apply blending
slc_id= _get_slice(4, ref_dim, i)
precip_blended[slc_id] = ws * precip_nowcast[slc_id] + (1 - ws) * precip_nwp[slc_id]
```

Reply:
This is indeed an elegant solution!